server/world/tick.go

```diff
@@ -52,6 +52,13 @@ func (t ticker) tick() {
 	}
 
 	rain, thunder, tick, tim := t.w.set.Raining, t.w.set.Thundering && t.w.set.Raining, t.w.set.CurrentTick, int(t.w.set.Time)
+	sleep := false
+	if t.w.set.RequiredSleepTicks > 0 {
+		t.w.set.RequiredSleepTicks--
+		if t.w.set.RequiredSleepTicks-1 <= 0 {
```

Comment:
Is this logic correct? You're decrementing the ticks and then checking if the tick below is <= 0?